### PR TITLE
docs: replace slice calls to map

### PR DESCRIPTION
**Pull request by just-boris**
As it was discussed in https://github.com/reactjs/redux/pull/1468, ES5 array methods like `map` and `filter` are recommended approach rather than array slicing.

### CODE REVIEW

**Comment by mxstbr** [comment]
IMO this looks much cleaner and easier to understand, thanks very much @just-boris!

I'm in a hurry right now, so I'll do a functional test soon to make sure everything still works as expected and then merge!

**Comment by just-boris** [comment]
I made changes only in Markdown files. So all code there wasn't actually executed during build.

@mxstbr How are gooing to test this?

**Comment by mxstbr** [comment]
Oh right, my bad, I didn't look at the code changes!

Looked through it now, LGTM, thanks very much @just-boris!

File: docs/Troubleshooting.md [hunk 1]
```diff
@@ -53,14 +53,15 @@ function todos(state = [], action) {
...
-      return [
-        ...state.slice(0, action.index),
-        // Copy the object before mutating
-        Object.assign({}, state[action.index], {
-          completed: true
-        }),
-        ...state.slice(action.index + 1)
-      ]
+      return state.map((todo, index) => {
+        if(index === action.index) {
```
Comment (gaearon) [inline docs/Troubleshooting.md]: Nit: we leave space after `if` pretty much everywhere else.
Would you mind fixing this as a separate PR?

Comment (just-boris) [inline docs/Troubleshooting.md]: No problem, attached it to #1478

**Pull request merged**